### PR TITLE
[1575] Allow validation tracking to be turned off locally

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/yi8K7QwO/1575-allow-validation-tracking-to-be-turned-off-locally

### Changes proposed in this pull request

* New setting added `Settings.track_validation_errors`
* Default is `true`

### Guidance to review

* Set `Settings.track_validation_errors` true/false
* Add new trainee, go to one of the forms and save without entering data
* Check `/system-admin/validation_errors`

### CODE REVIEW

**Comment by mooktakim** [comment]
[[S] Allow validation tracking to be turned off locally](https://trello.com/c/yi8K7QwO/1575-s-allow-validation-tracking-to-be-turned-off-locally)

File: config/settings.yml [hunk 1]
```diff
@@ -77,3 +77,5 @@ environment:
 
 slack:
   publish_register_alerts_channel: "#twd_publish_register_tech"
+
+track_validation_errors: true
```
Comment (defong) [inline config/settings.yml:81]: @darokel Did you wanted it off as default for `development.yml`?

File: spec/forms/trainee_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    it "saves validation errors" do
+      subject.valid?
+      expect(ValidationError.last.form_object).to eql("TestForm")
+      expect(ValidationError.count).to eql(1)
+
+      subject.valid?
+      expect(ValidationError.count).to eql(2)
+
+      # expect(subject.errors[:course_end_date]).to be_empty
```
Comment (defong) [inline spec/forms/trainee_form_spec.rb:34]: leftover?

File: spec/forms/trainee_form_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+require "rails_helper"
+
+class TestForm < TraineeForm
+  attr_accessor :test1, :test2, :test3
+  validates_presence_of :test1, :test2, :test3
+
+private
+
+  def compute_fields
```
Comment (defong) [inline spec/forms/trainee_form_spec.rb:11]: Wondering what this is...

**Review by defong — APPROVED** [review]
A few minor bits...

File: spec/forms/trainee_form_spec.rb (hunk 3 above)
Comment (mooktakim) [reply spec/forms/trainee_form_spec.rb:11]: TraineeForm needs that method implemented

File: config/settings.yml (hunk 1 above)
Comment (darokel) [reply config/settings.yml:81]: Yes I think it should be off by default and enabled explicitly

**Pull request merged**